bench-vortex/src/bin/tpch.rs

```diff
@@ -123,34 +123,34 @@ fn main() -> anyhow::Result<()> {
         _guard
     };
 
+    let formats = args.targets.iter().map(|t| t.format()).collect_vec();
+
     let runtime = new_tokio_runtime(args.threads);
 
+    let duckdb_resolved_path = ddb::get_executable_path(&args.duckdb_path);
+    if args.duckdb_path.is_none() && !args.skip_rebuild {
+        ddb::build_executable_path();
+    }
+
     let url = match args.use_remote_data_dir {
         None => {
-            let data_dir = match args.data_generator {
-                DataGenerator::Dbgen => {
-                    let db_gen_options =
-                        DBGenOptions::default().with_scale_factor(args.scale_factor);
-                    DBGen::new(db_gen_options).generate()?
-                }
-                DataGenerator::Duckdb => todo!("not implemented yet, will be support this soon"),
-                // TODO(joe) re-enable this once its correct.
-                // DataGenerator::Duckdb => {
-                //     generate_tpc(DuckdbTpcOptions::default().with_scale_factor(args.scale_factor))?
-                // }
-            };
+            for format in formats {
+                // Arrow uses csv
+                let format = if format == Format::Arrow {
```

Comment:
For patterns like this combinators make the code a bit more compact. Along the lines of 
```
(format == Format::Arrow).then().or()
```